src/lib/autoLauncher.ts

```diff
@@ -0,0 +1,141 @@
+      return true
+    } catch (error) {
+      console.error(error)
+      return false
+    }
+  },
+  getDirectory(): string { return '~/.config/autostart/'; },
+  getFilePath(appName: string): string { return `${this.getDirectory()}${appName}.desktop`; }
+}
+
+const macAL = {
+  async enable({ appName, appPath, hidden }: AutoLaunchParams): Promise<ChildReturnData> {
+    const isHiddenValue = hidden ? 'true' : 'false';
+    const properties = `{path:"${appPath}", hidden:${isHiddenValue}, name:"${appName}"}`;
+    console.log('properties', properties);
+    return native.execApplescriptCommand(`make login item at end with properties ${properties}`);
+  },
+  async disable(appName: string): Promise<ChildReturnData> {
+    return native.execApplescriptCommand(`delete login item \"${appName}\"`);
+  },
+  async isEnabled(appName: string): Promise<boolean> {
+    const response: ChildReturnData = await native.execApplescriptCommand('get the name of every login item');
+    const loginList = response?.stdout[0]?.split(', ') || []
+    const exists = loginList.includes(appName)
+    console.log('login Item Exists:', exists);
+    return exists
+  },
+
+}
+
+const winAL = {
+  subKey: "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run",
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async enable({ appName, appPath, hidden }: AutoLaunchParams): Promise<ChildReturnData> {
+    const returnVal = <ChildReturnData>{ stdout: [], stderr: [] }
+    const result = await native.winregSet(this.subKey, appName, appPath)
+    if (result.search('success') > -1) returnVal.stdout.push(result)
+    else returnVal.stderr.push(result)
+    return returnVal
+  },
+  async disable(appName: string): Promise<ChildReturnData> {
+    const returnVal = <ChildReturnData>{ stdout: [], stderr: [] }
+    const result = <string>(await native.winregDelete(this.subKey, appName))
+    if (result.search('success') > -1) returnVal.stdout.push(result)
+    else returnVal.stderr.push(result)
+    return returnVal
+  },
+  async isEnabled(appName: string): Promise<boolean> {
+    const result = <string>(await native.winregGet(this.subKey, appName))
+    console.log('isEnabled result:', result);
+    if (result.search('The system cannot find the file specified.') > -1) return false
+    else return true
+  }
+}
+
+
+export class AutoLauncher {
+  protected autoLauncher: osAL = nullAL
+  appName = 'app'
+  appPath = ''
+  enabled = false
+  async isEnabled(): Promise<boolean> {
+    const result = await this.autoLauncher.isEnabled(this.appName)
+    this.enabled = result
+    return result
+  }
+  async enable(): Promise<void | ChildReturnData> {
+    const child = await this.autoLauncher.enable({ appName: this.appName, appPath: this.appPath, hidden: false })
+    return child
+  }
+  async disable(): Promise<void | ChildReturnData> {
+    const child = this.autoLauncher.disable(this.appName)
+    return child
+  }
+  async init(): Promise<void> {
```

Comment:
The same flawed approach with initialization here, please either make sure it is impossible to instantiate uninitialized `AutoLauncher` or add TODO to fix that.